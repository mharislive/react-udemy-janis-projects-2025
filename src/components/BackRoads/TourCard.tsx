import { faMap } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

type TourCardProps = {
  content: string;
  image: string;
  date: string;
  title: string;
  country: string;
  duration: string;
  price: string;
};

const TourCard = (props: TourCardProps) => {
  const { content, image, date, title, country, duration, price } = props;

  return (
    <div className="flex flex-col flex-1 shadow hover:shadow-2xl hover:scale-105 transition duration-500">
      <div className="relative">
        <img src={image} alt={title} className="w-full" />
        <span className="absolute bottom-0 right-0 bg-cyan-300 px-2 py-1 capitalize">
          {date}
        </span>
      </div>
      <div className="py-5 px-6 bg-slate-100">
        <h3 className="font-bold tracking-widest mb-3 capitalize">{title}</h3>
        <p className="text-slate-500 mb-5">{content}</p>
        <div className="flex items-center justify-between text-cyan-500">
          <span className="capitalize">
            <FontAwesomeIcon icon={faMap} /> {country}
          </span>
          <span>{duration}</span>
          <span>{`From ${price}`}</span>
        </div>
      </div>
    </div>
  );
};

export default TourCard;
