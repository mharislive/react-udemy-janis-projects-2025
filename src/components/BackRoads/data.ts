import { faSocks, faTree, faWallet } from "@fortawesome/free-solid-svg-icons";
import tour1 from "../../assets/backroads/tour-1.jpeg";
import tour2 from "../../assets/backroads/tour-2.jpeg";
import tour3 from "../../assets/backroads/tour-3.jpeg";
import tour4 from "../../assets/backroads/tour-4.jpeg";

export const serviceData = [
  {
    title: "saving money",
    content:
      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Asperiores, officia.",
    icon: faWallet,
  },
  {
    title: "endless hiking",
    content:
      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Asperiores, officia.",
    icon: faTree,
  },
  {
    title: "amazing comfort",
    content:
      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Asperiores, officia.",
    icon: faSocks,
  },
];

export const toursdata = [
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour1,
    date: "august 26th, 2020",
    title: "Tibet Adventure",
    country: "china",
    duration: "6 days",
    price: "$2100",
  },
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour2,
    date: "october 1th, 2020",
    title: "best of java",
    country: "indonesia",
    duration: "11 days",
    price: "$1400",
  },
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour3,
    date: "september 15th, 2020",
    title: "explore hong kong",
    country: "hong kong",
    duration: "8 days",
    price: "$5000",
  },
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour4,
    date: "december 5th, 2019",
    title: "kenya highlights",
    country: "kenya",
    duration: "20 days",
    price: "$3300",
  },
];
