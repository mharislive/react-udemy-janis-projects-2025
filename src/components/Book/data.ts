import type { BookType } from "../../types/Book";

export const BOOKS: BookType[] = [
  {
    title: "Atomic Habits",
    author: "James Clear",
    image: "https://m.media-amazon.com/images/I/81F90H7hnML._SY425_.jpg",
    id: 1,
  },
  {
    title: "Word Power Made Easy",
    author: "Norman Lewis",
    image: "https://m.media-amazon.com/images/I/71DortY2MXL._SY425_.jpg",
    id: 2,
  },
  {
    title: "The Psychology of Money Paperback",
    author: "Morgan Housel",
    image: "https://m.media-amazon.com/images/I/71XEsXS5RlL._SY425_.jpg",
    id: 3,
  },
];
