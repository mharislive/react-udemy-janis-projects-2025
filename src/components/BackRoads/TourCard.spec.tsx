import { render, screen } from "@testing-library/react";
import { axe } from "vitest-axe";
import TourCard from "./TourCard";
import { toursdata } from "./data";

const mockProps = toursdata[0];

describe("BackRoads - TourCard", () => {
  it("renders the image with correct src and alt", () => {
    render(<TourCard {...mockProps} />);
    const img = screen.getByRole("img", { name: mockProps.title });
    expect(img).toHaveAttribute("src", mockProps.image);
    expect(img).toHaveAttribute("alt", mockProps.title);
  });

  it("displays the date badge", () => {
    render(<TourCard {...mockProps} />);
    expect(screen.getByText(mockProps.date)).toBeInTheDocument();
  });

  it("renders the title", () => {
    render(<TourCard {...mockProps} />);
    expect(
      screen.getByRole("heading", { name: mockProps.title })
    ).toBeInTheDocument();
  });

  it("renders the content/description", () => {
    render(<TourCard {...mockProps} />);
    expect(screen.getByText(mockProps.content)).toBeInTheDocument();
  });

  it("renders the country with map icon", () => {
    render(<TourCard {...mockProps} />);
    expect(screen.getByText(mockProps.country)).toBeInTheDocument();
    expect(document.querySelector("svg")).toBeInTheDocument();
  });

  it("renders the duration", () => {
    render(<TourCard {...mockProps} />);
    expect(screen.getByText(mockProps.duration)).toBeInTheDocument();
  });

  it("renders the price with 'From' prefix", () => {
    render(<TourCard {...mockProps} />);
    expect(screen.getByText(`From ${mockProps.price}`)).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(<TourCard {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have no accessibility violations", async () => {
    const { container } = render(<TourCard {...mockProps} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
