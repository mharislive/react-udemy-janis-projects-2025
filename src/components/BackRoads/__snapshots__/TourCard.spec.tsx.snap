// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BackRoads - TourCard > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col flex-1 shadow hover:shadow-2xl hover:scale-105 transition duration-500"
  >
    <div
      class="relative"
    >
      <img
        alt="Tibet Adventure"
        class="w-full"
        src="/src/assets/backroads/tour-1.jpeg"
      />
      <span
        class="absolute bottom-0 right-0 bg-cyan-300 px-2 py-1 capitalize"
      >
        august 26th, 2020
      </span>
    </div>
    <div
      class="py-5 px-6 bg-slate-100"
    >
      <h3
        class="font-bold tracking-widest mb-3 capitalize"
      >
        Tibet Adventure
      </h3>
      <p
        class="text-slate-500 mb-5"
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.
      </p>
      <div
        class="flex items-center justify-between text-cyan-500"
      >
        <span
          class="capitalize"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-map "
            data-icon="map"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M384 476.1L192 421.2l0-385.3L384 90.8l0 385.3zm32-1.2l0-386.5L543.1 37.5c15.8-6.3 32.9 5.3 32.9 22.3l0 334.8c0 9.8-6 18.6-15.1 22.3L416 474.8zM15.1 95.1L160 37.2l0 386.5L32.9 474.5C17.1 480.8 0 469.2 0 452.2L0 117.4c0-9.8 6-18.6 15.1-22.3z"
              fill="currentColor"
            />
          </svg>
           china
        </span>
        <span>
          6 days
        </span>
        <span>
          From $2100
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
